examples_test.go

```diff
@@ -0,0 +1,65 @@
+package spdx_go_test
+
+import (
+	"encoding/json"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/spdx/spdx-go-model/spdx_v3_0_1"
+)
+
+func TestImport(t *testing.T) {
+	spdx_v3_0_1.MakePerson()
+}
+
+func TestExample(t *testing.T) {
+	// Create a new Person
+	p := spdx_v3_0_1.MakePerson()
+
+	// Set the unique identifier (IRI) for the person
+	p.ID().Set("http://spdx.org/spdxdocs/person-example")
+
+	// Set the name for the person
+	p.Name().Set("Joshua")
+
+	// Create a creation info object
+	ci := spdx_v3_0_1.MakeCreationInfo()
+	ci.SpecVersion().Set("3.0.1")
+	ci.Created().Set(time.Now())
+
+	// Set the person as the creator of the SPDX data
+	ci.CreatedBy().AppendObj(p)
+
+	// Set the creation info as the creation info of the Person
+	p.CreationInfo().SetObj(ci)
+
+	// Add an e-mail address for the person
+	email := spdx_v3_0_1.MakeExternalIdentifier()
+	email.ExternalIdentifierType().Set(spdx_v3_0_1.ExternalIdentifierTypeEmail)
+	email.Identifier().Set("noreply@noreply.com")
+	p.ExternalIdentifier().AppendObj(email)
+
+	// Create a SPDX Document object
+	doc := spdx_v3_0_1.MakeSpdxDocument()
+	doc.ID().Set("http://spdx.org/spdxdocs/doc-example")
+	doc.CreationInfo().SetObj(ci)
```

Comment:
Similarly these one, as per above suggested to avoid getter.  
- Instead of `doc.CreationInfo.Set(value)`, use `doc.SetCreationInfo(ci)`, and
- `doc.ID().Set(value)`, by `doc.SetID(value)`

Refractor with:

```go
func (d *SpdxDocument) SetCreationInfo(ci CreationInfo) {
    d.CreationInfo = ci
}
``` 
and 
```go
func (d *SpdxDocument) SetID(id string) {
    d.SetID = id
}
```

Reply:
It can't be that simple, due to validation requirements (as stated, an important point of these bindings is to fail as early as possible if a bad value is assigned)

CreationInfo can be either a Object or a String IRI reference, so maybe something like:
```go
doc.SetCreationInfoObj(value)
doc.SetCreationInfoIRI("http://foo.bar/creationinfo")
doc.SetCreationInfo(creationInfoRef)
```
would be OK?